prow/plugins/trigger/generic-comment_test.go

```diff
@@ -508,17 +509,17 @@ func TestHandleGenericComment(t *testing.T) {
 						RegexpChangeMatcher: config.RegexpChangeMatcher{
 							RunIfChanged: "CHANGED2",
 						},
-						Context:      "pull-jib",
-						Trigger:      `/test (all|pull-jeb)`,
-						RerunCommand: `/test pull-jeb`,
+						Context:      "pull-jeb",
+						Trigger:      `(?m)^/test (?:.*? )?jeb(?: .*?)?$`,
+						RerunCommand: `/test jeb`,
 					},
 				},
 			},
-			ShouldBuild: true,
+			ShouldBuild: false,
```

Comment:
This change means that it will no longer be possible to force a `RunIfChanged` job to run with `/test foo` if the PR's changes don't match the regexp. That might be a problem for some use cases.
@BenTheElder @krzyzacy @munnerz  WDYT?

Reply:
What if we have a job that is run_if_changed, but which we sometimes still want to trigger manually on PRs that don't match the regexp? After the changes in this PR we would need to defined 2 copies of the job in the config, one with the run_if_changed and a copy of the job without run_if_changed.

Reply:
Is that something we do today? I thought today if it didn't match it wouldn't run, but it would leave a `Skipped` status

Reply:
No, today if you `/test foo` a RunIfChanged job that doesn't match it will still run.
e.g. https://github.com/kubernetes/test-infra/blob/58b41bfa9d9946c3d44b19ae3419290725ef1222/config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml#L56-L59
/test pull-test-infra-gubernator

Reply:
> pull-test-infra-gubernator Pending — Job triggered.

Reply:
Oh god

Reply:
To what extent is this a useful feature? ;)

Reply:
LOL @ gubernator test flaking

Reply:
I can implement this feature, the filters will just return `filtered, forced, defaults` for each job